File: backend/src/configs/env.ts
import dotenv from 'dotenv';

dotenv.config();

export const env = {
  NODE_ENV: process.env.NODE_ENV || 'development',
  PORT: parseInt(process.env.PORT || '5006', 10),
  HOST: process.env.HOST || '0.0.0.0',
  // Database configuration
  DB_TYPE: (process.env.DB_TYPE || 'mongodb') as 'mongodb',
  DB_HOST: process.env.DB_HOST || 'localhost',
  DB_PORT: parseInt(process.env.DB_PORT || '27017', 10),
  DB_NAME: process.env.DB_NAME || 'dallosh_analysis',
  DB_USER: process.env.DB_USER || undefined,
  DB_PASSWORD: process.env.DB_PASSWORD || undefined,

  DEFAULT_ADMIN_EMAIL: process.env.DEFAULT_ADMIN_EMAIL || 'admin@free.com',
  DEFAULT_ADMIN_PASSWORD: process.env.DEFAULT_ADMIN_PASSWORD || 'admin123',
  // Legacy MongoDB URI (for backward compatibility)
  MONGODB_URI: process.env.MONGODB_URI,
  JWT_SECRET: process.env.JWT_SECRET || 'your-secret-key-change-in-production',
  JWT_EXPIRES_IN: process.env.JWT_EXPIRES_IN || '7d',
  RABBITMQ_URL: process.env.RABBITMQ_URL || 'amqp://localhost:5672',
  STORAGE_PATH: process.env.STORAGE_PATH || '../../storage',
  RABBITMQ_TOPIC_TASKS: process.env.RABBITMQ_TOPIC_TASKS || 'tasks',
} as const;

File: microservices/auto_processing_datasets/src/configs/env.py
"""Environment variables configuration."""
import os
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

# Database configuration
DB_TYPE = os.getenv('DB_TYPE', 'mongodb')
DB_HOST = os.getenv('DB_HOST', 'localhost')
DB_PORT = int(os.getenv('DB_PORT', '27017'))
DB_NAME = os.getenv('DB_NAME', 'dallosh_analysis')
DB_USER = os.getenv('DB_USER', '')
DB_PASSWORD = os.getenv('DB_PASSWORD', '')

# RabbitMQ configuration
RABBITMQ_URL = os.getenv('RABBITMQ_URL', 'amqp://guest:guest@localhost:5672/')
RABBITMQ_TOPIC_TASKS = os.getenv('RABBITMQ_TOPIC_TASKS', 'tasks')  # Must match backend

# Storage paths
STORAGE_PATH = os.getenv('STORAGE_PATH', os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__)))), 'storage'))
STORAGE_DATASETS = os.getenv('STORAGE_DATASETS', os.path.join(STORAGE_PATH, 'datasets'))
STORAGE_CLEANED = os.getenv('STORAGE_CLEANED', os.path.join(STORAGE_PATH, 'cleaned'))
STORAGE_ANALYSED = os.getenv('STORAGE_ANALYSED', os.path.join(STORAGE_PATH, 'analysed'))

# LLM processing defaults
DEFAULT_PAGINATE_ROWS_LIMIT = int(os.getenv('DEFAULT_PAGINATE_ROWS_LIMIT', '500'))
DEFAULT_RETRY_REQUESTS = int(os.getenv('DEFAULT_RETRY_REQUESTS', '3'))
MAX_PAGINATE_ROWS_LIMIT = int(os.getenv('MAX_PAGINATE_ROWS_LIMIT', '1000'))
MAX_RETRY_REQUESTS = int(os.getenv('MAX_RETRY_REQUESTS', '5'))

